### ISSUE TITLE
limit numer of processes started

### ISSUE THREAD

**Issue by gsteenss** [rclone/rclone#1605]
Hi,

First of all: great tool!
I was wondering if it would be possible to limit the number of user processes that are created when syncing file to a remote?
I'm currently bumping into a limit on the hosting server account I would like to sync backups from to a Google drive, with the following command:

 ./rclone -v sync ~/sync-test.tgz  GoogleDrive:backups/
2017/08/16 07:42:28 INFO  : Google drive root 'backups': Modify window is 1ms
2017/08/16 07:42:28 ERROR : Ignoring --no-traverse with sync
2017/08/16 07:42:28 NOTICE: www: Can't follow symlink without -L/--copy-links
2017/08/16 07:42:28 NOTICE: access-logs: Can't follow symlink without -L/--copy-links
2017/08/16 07:42:28 INFO  : Google drive root 'backups': Waiting for checks to finish
2017/08/16 07:42:28 INFO  : Google drive root 'backups': Waiting for transfers to finish
runtime: failed to create new OS thread (have 33 already; errno=11)
runtime: may need to increase max user processes (ulimit -u)
fatal error: newosproc

Basically the hosting account has ulimit -u set to (only) 40, which is why it fails when trying to create another process.

If I were able to tell rclone to spawn a maximum 20 processes it might pass the limit and sucessfully sync the files, wouldn't mind if syncing was slower due to it really.

> What is your rclone version (eg output from `rclone -V`)

rclone v1.37 (same issue with beta: rclone-v1.37-093-gb510c70cβ-linux-amd64 )

> Which OS you are using and how many bits (eg Windows 7, 64 bit)
linux 64bit

> Which cloud storage system are you using? (eg Google Drive)

Google Drive

> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

./rclone -v sync ~/sync-test.tgz  GoogleDrive:backups/

**Comment by breunigs** [comment]
I read up on Go documentation a bit and it seems the language does not support this (e.g. https://github.com/golang/go/issues/14835#issuecomment-197407919). Not saying it's impossible, but probably a lot of effort.

For now, can you try to limit rclone's parallelism and see if that is enough? Try:
```
./rclone -v sync --checkers 1 --transfers 1 ~/sync-test.tgz GoogleDrive:backups/
```

**Comment by gsteenss** [comment]
Hi,  thanks for your reply.
I tried the '--transfers' params before, but even when adding '--checkers'  I get the same error when trying to sync.

**Comment by ncw** [comment]
How many CPUs has the host got?  By default the go runtime will make one thread per CPU - setting [GOMAXPROCS](https://golang.org/pkg/runtime/#hdr-Environment_Variables) will limit that.

    export GOMAXPROCS=1

As well as using `--checkers 1 --transfers 1` - see if that helps?

**Comment by animosity22** [comment]
This was answered. If you still have a question, please feel free to post on the forums.
